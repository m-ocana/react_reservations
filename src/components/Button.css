
button {
  border: none;
  background: none;
  cursor: pointer;
}

.close {
  float: right;
  font-size: 20px;
}

.close::before {
  content: "\2715";
}
