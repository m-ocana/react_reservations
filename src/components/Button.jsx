import React, { Component, PropTypes } from 'react'
import style from './Button.css'

export default class Button extends Component {

  static PropTypes = {
    className: PropTypes.string,
    clickHandler: PropTypes.func
  }

  static defaultProps = {
    className: '',
    clickHandler: () => {}
  }

  handleClick = (e) => {
    e.preventDefault()
    this.props.clickHandler()
  }

  render() {
    const { className } = this.props
    return (
      <button
        className={style[`${className}`]}
        onClick={this.handleClick}
      >
        { this.props.children }
      </button>
    )
  }
}
