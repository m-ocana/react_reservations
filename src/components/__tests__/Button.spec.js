import renderComponent from '../../helpers/test-component'
import Button from '../Button'

const renderButton = renderComponent(Button)
const initialProps = {
  className: 'button',
  clickHandler: jest.fn()
}

describe('<Button /> component', () => {
  describe('@renders', () => {
    it('should render dumb Button', () => {
      expect(renderButton().getTree())
        .toMatchSnapshot()
    })

    it('should render Button with initial props', () => {
      expect(renderButton(initialProps).getTree())
        .toMatchSnapshot()
    })

    it('should render Button with children', () => {
      expect(renderButton({
        ...initialProps,
        children: 'Im a button!'
      }).getTree()).toMatchSnapshot()
    })
  })

  describe('@events', () => {
    beforeEach(() => jest.resetAllMocks())

    describe('on button click', () => {
      it('should trigger clickHandler', () => {
        const { wrapper, instance } = renderButton(initialProps)
        wrapper.find('.button').simulate('click', { preventDefault: jest.fn() })
        expect(instance.props.clickHandler).toHaveBeenCalled()
      })
    })
  })
})
