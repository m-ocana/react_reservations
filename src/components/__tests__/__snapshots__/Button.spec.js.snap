// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component @renders should render Button with children 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Im a button!
</button>
`;

exports[`<Button /> component @renders should render Button with initial props 1`] = `
<button
  className="button"
  onClick={[Function]}
/>
`;

exports[`<Button /> component @renders should render dumb Button 1`] = `
<button
  className=""
  onClick={[Function]}
/>
`;
